File: python/p0242_valid_anagram/solutions/brute_force.py
# Time complexity: O(n^2)
# Space complexity: O(1)


class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        if len(s) != len(t):
            return False

        for letter in s:
            if s.count(letter) != t.count(letter):
                return False

        return True

File: rust/p0242_valid_anagram/src/brute_force.rs
// Time complexity: O(n^2)
// Space complexity: O(n)

pub struct Solution;

impl Solution {
    pub fn is_anagram(s: String, t: String) -> bool {
        if s.len() != t.len() {
            return false;
        }

        for letter in s.chars() {
            if s.matches(letter).count() != t.matches(letter).count() {
                return false;
            }
        }

        true
    }
}
